lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -259,14 +261,42 @@ def update(self, subsection, block_structure=None):
         self._log_event(log.warning, u"update, subsection: {}".format(subsection.location))
 
         block_structure = self._get_block_structure(block_structure)
-        subsection_grade = SubsectionGrade(subsection, self.course)
-        subsection_grade.init_from_structure(
-            self.student, block_structure, self._submissions_scores, self._csm_scores
+        calculated_grade = SubsectionGrade(subsection, self.course).init_from_structure(
+            self.student, block_structure, self._submissions_scores, self._csm_scores,
         )
 
-        grade_model = subsection_grade.update_or_create_model(self.student)
-        self._update_saved_subsection_grade(subsection.location, grade_model)
-        return subsection_grade
+        should_update_grade = True
+        if only_if_higher:
+            calculated_grade, should_update_grade = self._choose_grade_to_update(
+                subsection, block_structure, calculated_grade,
+            )
+
+        if should_update_grade:
+            grade_model = calculated_grade.update_or_create_model(self.student)
+            self._update_saved_subsection_grade(subsection.location, grade_model)
+
+        return calculated_grade
+
+    def _choose_grade_to_update(self, subsection, block_structure, new_calculated_grade):
+        """
+        Returns a tuple of the chosen grade and whether it needs to be updated.
+        """
+        try:
+            grade_model = PersistentSubsectionGrade.read_grade(self.student.id, subsection.location)
+            orig_subsection_grade = SubsectionGrade(subsection, self.course).init_from_model(
+                self.student, grade_model, block_structure, self._submissions_scores, self._csm_scores,
```

Comment:
Yeah, it's not strictly required - our convention is just not to mix the 2 formatting strategies: either all in one line or each in separate lines.
I'm gonna keep as is, for now, since it's consistent with all the other `init_from_model` and `init_from_structure` calls in this file.